spack-ext/lib/jcsda-emc/spack-stack/stack/stack_env.py

```diff
@@ -263,11 +263,7 @@ def write(self):
             target_compiler = f"%{self.compiler}"
             for i in range(len(definitions)):
                 if "compilers" in definitions[i]:
-                    j = len(definitions[i]["compilers"])-1
-                    while j>=0:
-                        if not definitions[i]["compilers"][j] == target_compiler:
-                            definitions[i]["compilers"].pop(j)
-                        j -= 1
+                    definitions[i] = {"compilers": [target_compiler]}
```

Comment:
This is so much easier; do you remember why we put that complicated logic in place originally (I don't)?

Reply:
No I don't remember... Maybe something to do with supporting multiple compilers in an environment..?

Reply:
Oh right. This is something we need to test before we merge. The Intel or oneAPI environments require some packages to be built with gcc, therefore the `compilers` line in the spack config must contain both compilers. See https://github.com/JCSDA/spack-stack/blob/a8bb4fb77b42364bcf9ca06c91a8e77e3b224ff3/configs/sites/tier1/nautilus/packages_intel.yaml#L3

We need to make sure that this still works (i.e. a spack module [tcl|lmod] refresh followed by spack stack setup-meta-modules still works as expected. I think it does, but better to check.

Reply:
Sure. I think we would expect it to work though, on account of there being only one value for target_compiler (based on self.compiler)? Unless users can specify multiple...